modules/reactor/src/meshgenerators/CoarseMeshExtraElementIDGenerator.C

```diff
@@ -0,0 +1,154 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CoarseMeshExtraElementIDGenerator.h"
+
+registerMooseObject("ReactorApp", CoarseMeshExtraElementIDGenerator);
+
+#include "libmesh/enum_point_locator_type.h"
+#include "libmesh/elem.h"
+#include "libmesh/mesh_serializer.h"
+
+InputParameters
+CoarseMeshExtraElementIDGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+  params.addRequiredParam<MeshGeneratorName>(
+      "input", "Name of an existing mesh generator to which we assign coarse element IDs");
+  params.addRequiredParam<MeshGeneratorName>(
+      "coarse_mesh", "Name of an existing mesh generator as the coarse mesh");
+  params.addRequiredParam<std::string>("coarse_element_id_name",
+                                       "Name for the coarse element ID on the input mesh");
+  params.addParam<std::string>(
+      "coarse_mesh_extra_element_id",
+      "Name for the extra element ID on the coarse mesh (default to element ID)");
```

Comment:
I love the new capability you just added, but now that first name is confusing. 

coarse_element_id_name is actually the name of the extra element integer on the input mesh. 
```suggestion
  params.addRequiredParam<std::string>("extra_element_id_name",
                                       "Name for the extra element ID that is added to the input mesh");
  params.addParam<std::string>(
      "coarse_mesh_extra_element_id",
      "Name for the extra element ID that copied from the coarse mesh (default to element ID)");
```